packages/vsce/src/commands/inspectResourceCommandUtils.ts

```diff
@@ -101,3 +142,51 @@ function getResourceType(resourceName: string): IResourceMeta<IResource> {
   }
   return undefined;
 }
+
+async function selectResourceType(): Promise<IResourceMeta<IResource> | undefined> {
+  const choice = await getChoiceFromQuickPick(CICSMessages.CICSSelectResourceType.message, SupportedResources.resources);
+
+  if (choice) {
+    return getResourceType(choice.label);
+  }
+
+  return undefined;
+}
+
+async function selectResource(resourceType: IResourceMeta<IResource>): Promise<string | undefined> {
+  return getEntryFromInputBox(resourceType);
+}
+
+async function getChoiceFromQuickPick(
+  placeHolder: string,
+  items: string[]
+): Promise<QuickPickItem | undefined> {
+  const qpItems: QuickPickItem[] = [...items.map((item) => ({ label: item}))];
+
+  const quickPick = Gui.createQuickPick();
+  quickPick.items = qpItems;
+  quickPick.placeholder = l10n.t(placeHolder);
+  quickPick.ignoreFocusOut = true;
+  quickPick.show();
+  const choice = await Gui.resolveQuickPick(quickPick);
+  quickPick.hide();
+  return choice;
+}
+
+async function getEntryFromInputBox(resourceType: IResourceMeta<IResource>): Promise<string | undefined> {
+  const options: InputBoxOptions = {
+    prompt: CICSMessages.CICSEnterResourceName.message,
+    value: "",
+    validateInput: function(value: string): string {
+      if ((resourceType === TransactionMeta && value.length > 4) ||
+        (resourceType !== TransactionMeta && value.length > 8)) {
+        return CICSMessages.CICSInvalidResourceName.message;
```

Comment:
Like we discussed, we could do something smarter here with the message, like telling them why it's invald.

`Invalid CICS Resource name. Maximum length is 8 characters`

<img width="311" height="76" alt="Image" src="https://github.com/user-attachments/assets/aa9f9547-91dd-405f-948c-e30f71a97012" />